Units test fail on big-endian architecture

**What doesn't work?**
We are trying to migrate to version 0.9.9 of pysdl2 (that we sync from Debian) for the next development release of Ubuntu.
We now run the pytest suite in our autopkgtest (we didn't do it before) and we noticed failures on our s390x (https://en.wikipedia.org/wiki/IBM_System/390) architecture, which is the only big-endian architecture we have.

The following tests are failing:
```
FAILED sdl2/test/pixels_test.py::TestSDLPixels::test_SDL_MasksToPixelFormatEnum
FAILED sdl2/test/pixels_test.py::TestSDLPixels::test_SDL_PixelFormatEnumToMasks
FAILED sdl2/test/sdl2ext_array_test.py::TestSDL2ExtArray::test_CTypesView__doublebytes
FAILED sdl2/test/sdl2ext_array_test.py::TestSDL2ExtArray::test_CTypesView__quadbytes
FAILED sdl2/test/sdl2ext_array_test.py::TestSDL2ExtArray::test_create_array
FAILED sdl2/test/sdl2ext_draw_test.py::TestSDL2ExtDraw::test_fill - assert False
FAILED sdl2/test/sdl2ext_draw_test.py::TestSDL2ExtDraw::test_line - assert False
FAILED sdl2/test/sdl2ext_pixelaccess_test.py::TestSDL2ExtPixelAccess::test_pixels3d
```
while everything seems fine on our other architectures.

I investigated the failures in `sdl2/test/pixels_test.py`:
```
        for fmt, expected in formats:
            bpp, r, g, b, a = expected
            enum = sdl2.SDL_MasksToPixelFormatEnum(bpp, r, g, b, a)
>           assert enum == fmt
E           assert 376840196 == 373694468

sdl2/test/pixels_test.py:153: AssertionError

            assert sdl2.SDL_GetError() == b""
            assert ret == SDL_TRUE
>           assert [x.value for x in (bpp, r, g, b, a)] == expected
E           assert [32, 42781900...0, 65280, 255] == [32, 255, 652...0, 4278190080]
E             At index 1 diff: 4278190080 != 255
E             Use -v to get the full diff

sdl2/test/pixels_test.py:174: AssertionError
```
The code contains:
```python
RGBA32 = [0xFF000000, 0x00FF0000, 0x0000FF00, 0x000000FF]
if sys.byteorder == "big":
    RGBA32.reverse()
```
and we fail when comparing the result of `SDL_MasksToPixelFormatEnum(bpp, R, G, B, A)` with `SDL_PIXELFORMAT_RGBA8888`
Calling the `.reverse()` seems wrong in this context since it would make us pass different RGBA values to SDL_MasksToPixelFormatEnum. Therefore, the SDL function returns the "wrong constant"
```c
[...]
        if (Rmask == 0xFF000000 &&
            Gmask == 0x00FF0000 &&
            Bmask == 0x0000FF00 &&
            Amask == 0x000000FF) {
            return SDL_PIXELFORMAT_RGBA8888; /* This is the value we get in Little-Endian */
        }
        if (Rmask == 0x000000FF &&
            Gmask == 0x0000FF00 &&
            Bmask == 0x00FF0000 &&
            Amask == 0xFF000000) {
            return SDL_PIXELFORMAT_ABGR8888; /* This is the value we get in Big-Endian */
        }
[...]
```

Removing the code that calls `.reverse()` for big-endian architectures seems to solve the problem for s390x but I would appreciate a second opinion in case I overlooked something.

Furthermore, the other failures seem unfortunately more complex to resolve.
I'd be interested to know if somebody has successfully run the tests on big-endian architectures "recently"?
Are we aware of any "recent" change in libsdl in the way endianness is handled, that could explain this regression? I tend to believe, that the `if sys.byteorder ...` statements have been put there at some point to fix the tests.

If you're interested in the full autopkgtest report, it is available at https://autopkgtest.ubuntu.com/results/autopkgtest-kinetic/kinetic/s390x/p/pysdl2/20220509_100317_74a0e@/log.gz

I would appreciate your help!

Thanks!

**How To Reproduce**
On a big-endian-based architecture, run the following command:
```sh
$ python3 -m pytest sdl2/test
```

**Platform (if relevant):**
 - OS: Linux s390x
 - Python Version: 3.10
 - SDL2 Version: [2.0.20
 - Using pysdl2-dll: No

**Additional context**
Add any other context about the problem here.

--- a/doc/news.rst
+++ b/doc/news.rst
@@ -49,6 +49,7 @@ Fixed Bugs:
   through the ``sdl2`` namespace).
 * Updated ``sdl2.dll.version`` to better handle SDL2's new versioning format
   and fixed unit tests accordingly (issue #228).
+* Fixed various unit tests on big-endian platforms (PR #232).
 
 
 0.9.11

--- a/sdl2/test/pixels_test.py
+++ b/sdl2/test/pixels_test.py
@@ -9,9 +9,6 @@ from sdl2.stdinc import Uint8, Uint16, Uint32
 
 RGBA32 = [0xFF000000, 0x00FF0000, 0x0000FF00, 0x000000FF]
 RGBX32 = [0xFF000000, 0x00FF0000, 0x0000FF00, 0]
-if sys.byteorder == "big":
-    RGBA32.reverse()
-    RGBX32.reverse()
 
 
 # Test custom macros

--- a/sdl2/test/pixels_test.py
+++ b/sdl2/test/pixels_test.py
@@ -7,8 +7,8 @@ from sdl2 import SDL_Init, SDL_Quit, SDL_INIT_EVERYTHING, SDL_TRUE
 from sdl2 import SDL_Color
 from sdl2.stdinc import Uint8, Uint16, Uint32
 
-RGBA32 = [0xFF000000, 0x00FF0000, 0x0000FF00, 0x000000FF]
-RGBX32 = [0xFF000000, 0x00FF0000, 0x0000FF00, 0]
+RGBA8888 = [0xFF000000, 0x00FF0000, 0x0000FF00, 0x000000FF]
+RGBX8888 = [0xFF000000, 0x00FF0000, 0x0000FF00, 0]
 
 
 # Test custom macros
@@ -140,8 +140,8 @@ def test_SDL_GetPixelFormatName():
 
 def test_SDL_MasksToPixelFormatEnum():
     formats = [
-        (sdl2.SDL_PIXELFORMAT_RGBA8888, [32] + RGBA32),
-        (sdl2.SDL_PIXELFORMAT_RGBX8888, [32] + RGBX32),
+        (sdl2.SDL_PIXELFORMAT_RGBA8888, [32] + RGBA8888),
+        (sdl2.SDL_PIXELFORMAT_RGBX8888, [32] + RGBX8888),
         (sdl2.SDL_PIXELFORMAT_INDEX1MSB, [1, 0, 0, 0, 0]),
     ]
     for fmt, expected in formats:
@@ -154,8 +154,8 @@ def test_SDL_MasksToPixelFormatEnum():
 
 def test_SDL_PixelFormatEnumToMasks():
     formats = [
-        (sdl2.SDL_PIXELFORMAT_RGBA8888, [32] + RGBA32),
-        (sdl2.SDL_PIXELFORMAT_RGBX8888, [32] + RGBX32),
+        (sdl2.SDL_PIXELFORMAT_RGBA8888, [32] + RGBA8888),
+        (sdl2.SDL_PIXELFORMAT_RGBX8888, [32] + RGBX8888),
         (sdl2.SDL_PIXELFORMAT_INDEX1LSB, [1, 0, 0, 0, 0]),
         (sdl2.SDL_PIXELFORMAT_INDEX1MSB, [1, 0, 0, 0, 0]),
         (sdl2.SDL_PIXELFORMAT_UNKNOWN, [0, 0, 0, 0, 0]),

--- a/sdl2/test/sdl2ext_array_test.py
+++ b/sdl2/test/sdl2ext_array_test.py
@@ -5,6 +5,8 @@ import struct
 import pytest
 from sdl2.ext import array as sdlextarray
 
+BIG_ENDIAN = sys.byteorder == "big"
+
 singlebyteseq = [x for x in range(0x100)]
 doublebyteseq = [x for x in range(0x10000)]
 quadbyteseq = [
@@ -92,7 +94,7 @@ def create_64b(seq, size, offset):
                     seq[offset + 2] << 16 |
                     seq[offset + 3])
         elif size == 4:
-            return (seq[offset] << 32 | seq[offset])
+            return (seq[offset] << 32 | seq[offset + 1])
 
 def lobyte16(val):
     return val & 0x00FF
@@ -187,12 +189,14 @@ class TestExtCTypesView(object):
             offset = 0
             cnt = 0
             for val in singlebytes.to_bytes():
+                hi = hibyte16(doublebyteseq[offset])
+                lo = lobyte16(doublebyteseq[offset])
                 if cnt > 0:
-                    assert val == hibyte16(doublebyteseq[offset])
+                    assert val == (lo if BIG_ENDIAN else hi)
                     cnt = 0
                     offset += 1
                 else:
-                    assert val == lobyte16(doublebyteseq[offset])
+                    assert val == (hi if BIG_ENDIAN else lo)
                     cnt += 1
 
             offset = 0
@@ -232,12 +236,14 @@ class TestExtCTypesView(object):
             cnt = 0
             offset = 0
             for val in singlebytes.to_uint16():
+                hi = hibytes32(quadbyteseq[offset])
+                lo = lobytes32(quadbyteseq[offset])
                 if cnt > 0:
-                    assert val == hibytes32(quadbyteseq[offset])
+                    assert val == (lo if BIG_ENDIAN else hi)
                     cnt = 0
                     offset += 1
                 else:
-                    assert val == lobytes32(quadbyteseq[offset])
+                    assert val == (hi if BIG_ENDIAN else lo)
                     cnt += 1
 
             offset = 0
@@ -367,9 +373,9 @@ def test_create_array():
         if i == 1:
             assert parr[0] == 0x0
         elif i == 2:
-            assert parr[0] == 0x0100
+            assert parr[0] == struct.unpack("=H", barr[0:2])[0]
         elif i == 4:
-            assert parr[0] == 0x03020100
+            assert parr[0] == struct.unpack("=I", barr[0:4])[0]
     for i in (0, 3, 5, 6, 7, 9, 10, 12, "test"):
         with pytest.raises(TypeError):
             sdlextarray.create_array(barr, i)

--- a/sdl2/test/sdl2ext_draw_test.py
+++ b/sdl2/test/sdl2ext_draw_test.py
@@ -16,7 +16,7 @@ except:
 
 @pytest.fixture
 def testsurf(with_sdl):
-    sf = SDL_CreateRGBSurface(0, 10, 10, 32, 0, 0, 0, 0)
+    sf = _create_surface((10, 10), fmt="RGBA32")
     assert SDL_GetError() == b""
     yield sf
     SDL_FreeSurface(sf)

--- a/sdl2/test/sdl2ext_image_test.py
+++ b/sdl2/test/sdl2ext_image_test.py
@@ -47,6 +47,10 @@ if _HASSDLIMAGE:
 # during tests
 skip_color_check = ['gif', 'jpg', 'lbm', 'pbm', 'pgm', 'svg', 'webp']
 
+# Skip ICO and CUR tests on big-endian, since they don't seem to work yet
+if sys.byteorder == "big":
+    skip_color_check += ['ico', 'cur']
+
 # SDL 2.0.10 has a bug that messes up converting surfaces with transparency
 if sdl2.dll.version == 2010:
     skip_color_check.append('xpm')

--- a/sdl2/test/sdl2ext_pixelaccess_test.py
+++ b/sdl2/test/sdl2ext_pixelaccess_test.py
@@ -30,7 +30,7 @@ def imgsurf(with_sdl):
 
 @pytest.fixture
 def rgbasurf(imgsurf):
-    rgba = pixels.SDL_PIXELFORMAT_ABGR8888
+    rgba = pixels.SDL_PIXELFORMAT_RGBA32
     rgbasurf = surface.SDL_ConvertSurfaceFormat(imgsurf.contents, rgba, 0)
     yield rgbasurf
     surface.SDL_FreeSurface(rgbasurf)

--- a/sdl2/test/sdl2ext_surface_test.py
+++ b/sdl2/test/sdl2ext_surface_test.py
@@ -2,6 +2,7 @@ import pytest
 from sdl2.surface import SDL_CreateRGBSurface, SDL_FreeSurface
 from sdl2.rect import SDL_Rect
 from sdl2.ext.draw import prepare_color, fill
+from sdl2.ext.surface import _create_surface
 from sdl2 import ext as sdl2ext
 
 try:
@@ -14,7 +15,7 @@ except:
 @pytest.mark.skipif(not _HASNUMPY, reason="Numpy not available")
 def test_subsurface(with_sdl):
     # Initialize colour and surface/view
-    sf = SDL_CreateRGBSurface(0, 10, 10, 32, 0, 0, 0, 0)
+    sf = _create_surface((10, 10), fmt="RGBA32")
     WHITE = prepare_color((255, 255, 255), sf)
 
     # Test creation of subsurface from parent

